src/MagnumPlugins/ObjImporter/ObjImporter.cpp

```diff
@@ -115,387 +372,575 @@ ObjImporter::~ObjImporter() = default;
+                //TODO C++17
+                ensureGroup();
+                meshData = &group->meshDataForMaterial(materialIndex);
+            } // else: usemtl did not result in material switch, no need to create new mesh
+
+        /* Ignore unsupported keywords, error out on unknown keywords */
+        } else {
+            Warning() << "Trade::ObjImporter::parse(): unknown keyword:" << keyword;
+        }
+
+        if(sectionEnd) {
+            finishSection();
         }
 
         /* Ignore the rest of the line */
-        ignoreLine(*_file->in);
+        line = pos = skipWhitespaces(ignoreLine(pos));
     }
 
-    /* Set end of the last object */
-    _file->in->clear();
-    _file->in->seekg(0, std::ios::end);
-    std::get<1>(_file->meshes.back()) = _file->in->tellg();
+    finishSection();
+    finishObject();
+
+    int i = _state->meshes.size();
+    for(auto name : _state->meshlessObjects) {
+        _state->meshlessObjectIds[name] = i++;
+    }
 }
 
-UnsignedInt ObjImporter::doMesh3DCount() const { return _file->meshes.size(); }
+void ObjImporter::parseMaterialLibrary(const ArrayView<char> libname) {
+    std::string filename = _fileRoot + std::string(libname.data(), libname.size());
 
-Int ObjImporter::doMesh3DForName(const std::string& name) {
-    const auto it = _file->meshesForName.find(name);
-    return it == _file->meshesForName.end() ? -1 : it->second;
-}
+    /* Open file */
+    if(!Utility::Directory::fileExists(filename)) {
+        Error() << "Trade::AbstractImporter::parseMaterialLibrary(): cannot open file" << filename;
+        return;
+    }
 
-std::string ObjImporter::doMesh3DName(UnsignedInt id) {
-    return _file->meshNames[id];
-}
+    Containers::Array<char> contents = Utility::Directory::read(filename);
+    ArrayView<char> pos = contents;  /* points to current character in line */
 
-std::optional<MeshData3D> ObjImporter::doMesh3D(UnsignedInt id) {
-    /* Seek the file, set mesh parsing parameters */
-    std::streampos begin, end;
-    UnsignedInt positionIndexOffset, textureCoordinateIndexOffset, normalIndexOffset;
-    std::tie(begin, end, positionIndexOffset, textureCoordinateIndexOffset, normalIndexOffset) = _file->meshes[id];
-    _file->in->seekg(begin);
+    ObjMaterial* mat = nullptr;
 
-    std::optional<MeshPrimitive> primitive;
-    std::vector<Vector3> positions;
-    std::vector<std::vector<Vector2>> textureCoordinates;
-    std::vector<std::vector<Vector3>> normals;
-    std::vector<UnsignedInt> positionIndices;
-    std::vector<UnsignedInt> textureCoordinateIndices;
-    std::vector<UnsignedInt> normalIndices;
+    while(!pos.empty()) {
 
-    try { while(_file->in->good() && _file->in->tellg() < end) {
-        /* Ignore comments */
-        if(_file->in->peek() == '#') {
-            ignoreLine(*_file->in);
+        /* Comment line */
+        if(pos[0] == '#') {
+            pos = ignoreLine(pos);
+            pos = skipWhitespaces(pos);
```

Comment:
Again, no, the ignore Whitespaces ignores words before the keyword (`nextWord()` expects first character to be non-whitespace, otherwise will return empty string, indicating there is no word at the beginning), versus the other one skips whitespaces *after* the keyword.